### PR TITLE
better use of cancellationToken

### LINKED ISSUE #179: CancellationToken usage in TimeoutsPoller
Any reason we dont pass through the CancellationToken on these lines

 * https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/develop/src/Transport/DelayDelivery/TimeoutsPoller.cs#L163
 * https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/develop/src/Transport/DelayDelivery/TimeoutsPoller.cs#L170
 * ~https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/develop/src/Transport/DelayDelivery/TimeoutsPoller.cs#L44~

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #179

also fixes nullreference exception in TimeoutEntity

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/azure-maintainers i think this is ready for a review

File: src/Transport/DelayDelivery/TimeoutsPoller.cs [hunk 1]
```diff
@@ -38,10 +38,12 @@ public TimeoutsPoller(string connectionString, Dispatcher dispatcher, string tab
...
 
-        public async Task Start(ReadOnlySettings settings, CancellationToken token)
+        public async Task Start(ReadOnlySettings settings, CancellationToken cancellationToken)
         {
-            await Init(settings).ConfigureAwait(false);
-            timeoutPollerTask = Task.Run(() => Poll(token));
+            await Init(settings, cancellationToken).ConfigureAwait(false);
+            // No need to pass token to run. to avoid when token is cancelled the task changing into 
+            // the cancelled state and when awaited while stopping rethrow the cancelled exception
+            timeoutPollerTask = Task.Run(() => Poll(cancellationToken));
```
Comment (boblangley) [inline src/Transport/DelayDelivery/TimeoutsPoller.cs]: I think we implemented here where we didn't intend to per https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/179#issuecomment-276587706

Comment (SimonCropp) [inline src/Transport/DelayDelivery/TimeoutsPoller.cs]: @boblangley i didnt actually add a cancellationToken there, i just renamed the variable. my question to @danielmarbach was about passing a token to `Task.Run` not `Poll` (which already existed)

Comment (SimonCropp) [inline src/Transport/DelayDelivery/TimeoutsPoller.cs]: but i did add the comment to remove the ambiguity for the next person

Comment (boblangley) [inline src/Transport/DelayDelivery/TimeoutsPoller.cs]: I misread it, thanks.

Comment (danielmarbach) [inline src/Transport/DelayDelivery/TimeoutsPoller.cs]: You could also pass in `CancellationToken.None` which is clearer IMHO

**Pull request merged**